worker/uniter/relation/statetracker.go

```diff
@@ -181,6 +195,8 @@ func (r *relationStateTracker) joinRelation(rel Relation) (err error) {
 			// Should this be a different error? e.g. resolver.ErrAborted, that
 			// Loop translates into ErrLoopAborted?
 			return resolver.ErrLoopAborted
+		case <-time.After(30 * time.Second):
```

Comment:
I don't imagine this loop will spin much but this could blow out to a lot of leaked timers+channels.

Reply:
The expectation is that the watcher *always* triggers with the initial event immediately.
There's no expectation that this will ever be timed out unless there's a terrible bug in the watcher. Nonetheless, I've tweaked to return an error if the timeout ever happens, which will bounce the worker and things can re-sync.